### ISSUE TITLE
Setting "option allow_alias = true" causes message compilation to fail

### ISSUE THREAD

**Issue by bb4242** [tomas-abrahamsson/gpb#28]
Trying to compile the following message with gpb fails:

```
message FailingTest{
    enum myEnum
    {
    option allow_alias = true;
    ENUM1 = 1;
    ENUM2 = 1;
    }
}
```
In case it's relevant, I'm compiling this message in a project that uses rebar, with a rebar.config as follows:
```
{proto_compiler, gpb}.
{gpb_opts, [maps, include_as_lib, use_packages, {verify, always}]}.
{deps,
 [
  {gpb, "", {git, "https://github.com/tomas-abrahamsson/gpb.git", {tag, "3.17.7"}}}
 ]
}.
```
The error message is:
```
$ ./rebar compile         
==> gpb (compile)
Compiled src/gpb.erl
==> messages (compile)
...messages/proto/FailingTest.proto:4: syntax error before: option
ERROR: Failed to compile src/messages/proto/FailingTest.proto
Compiling src/messages/proto/FailingTest.proto failed:
ERROR: compile failed while processing: rebar_abort
```

I'm not sure if `allow_alias` is a supported option or not.  The gpb readme states:
```
gpb reads but ignores or throws away:
options other than 'packed' or 'default'
custom options
```
From this, I would expect gpb to ignore the allow_alias option, rather than failing to compile.  As a side note, if I remove the option, the message does compile with a warning:
```
src/FailingTest.erl:180: Warning: this clause cannot match because a previous clause at line 179 always matches
```
This makes sense, because the lines in question are the `'enum_symbol_by_value_FailingTest.myEnum'` function, which looks up the enum symbol by value.

I'm compiling messages that I don't have direct control over, so I'd actually prefer if gpb would just ignore the allow_alias option, rather than failing to compile.  Is it possible to change this behavior?  Thanks! :)

**Comment by tomas-abrahamsson** [comment]
I see. I think options at the time of writing that sentence about being ignored, were maybe only allowed after an enumerator, in any case I didn't foresee it at that position.

Ignoring the option at encoding would be rather easy, I think. On decoding, I suppose a reasonable approach could be to decode to the first enumerator symbol, `ENUM1` in this case.

I'll take a look at it, hopefully during the weekend.

**Comment by bb4242** [comment]
Yeah, I agree with your points: ignoring the option at encoding, and decoding to the first enumerator symbol, sounds like very reasonable behavior.  Thanks for the library - it's really handy!

**Comment by tomas-abrahamsson** [comment]
I have pushed a solution, to the branch `enum-option-allow_alias`. My intention is to try to also include the other issue with maps and optional fields, soon. But if that should get delayed, I will merge this to master, so that at least this won't get held up. (My plan is of course that 'soon' will happens soon, but there's always the unexpected, and meanwhile there's a solution on a branch in case you want to test it out.)

**Comment by bb4242** [comment]
Thanks very much! :)  I did a quick test of the new `enum-option-allow_alias` branch, and that certainly looks to solve the issue I had.

**Comment by tomas-abrahamsson** [comment]
Good to hear, thanks for quick feedback!

**Comment by bb4242** [comment]
No problem, thanks for the quick fix. :)

**Comment by tomas-abrahamsson** [comment]
This is now merged to master and in 3.17.8.

**Comment by bb4242** [comment]
Thanks!
